cpp/src/parquet/encoding.h

```diff
@@ -72,10 +181,24 @@ class TypedEncoder : virtual public Encoder {
 
   virtual void Put(const T* src, int num_values) = 0;
 
+  virtual void Put(const std::vector<T>& src, int num_values = -1);
+
   virtual void PutSpaced(const T* src, int num_values, const uint8_t* valid_bits,
                          int64_t valid_bits_offset) = 0;
 };
 
+template <typename DType>
+void TypedEncoder<DType>::Put(const std::vector<T>& src, int num_values) {
+  if (num_values == -1) {
+    num_values = static_cast<int>(src.size());
+  }
+  Put(src.data(), num_values);
+}
+
+template <>
+inline void TypedEncoder<BooleanType>::Put(const std::vector<bool>& src, int num_values) {
+}
+
```

Comment:
An incomplete thought here?

Reply:
No, this is overridden later by `PlainEncoder<BooleanType>`. It wasn't possible to provide a default implementation for non boolean but make the method pure virtual for boolean. (Note that non boolean encoders don't override `Put(vector<...`.) I can't write a non empty default impl for boolean because I don't have access to `PlainEncoder<BooleanType>::bit_writer_`.

Reply:
I will add a comment here so future readers are not confused